### ISSUE TITLE
Multi-GPU training

### ISSUE THREAD

**Issue by mr-nlp** [UKPLab/sentence-transformers#556]
Does sentence-transformers package supports multi-gpu training? 

I think as of now, we can use sentence-transformers package to train bert based models on ALLNLI / STS datasets using only a single GPU. (correct me if i'm wrong)

@nreimers

**Comment by nreimers** [comment]
Hi @mr-nlp 
Currently is it not implemented. I tried to add DataParallel, but the training speed decreased with more GPUs compared to a single GPU.

**Comment by mr-nlp** [comment]
@nreimers  Thanks for the quick response. Is there any plans to include multi-gpu training support in sentence-transformers package in any other way in future?

**Comment by nreimers** [comment]
Hi @mr-nlp 
It is planned, but unclear when it will happen. As mentioned, the easy and straightforward solution with DataParallel did not yield a speedup

**Comment by mr-nlp** [comment]
Can you please tell me the configuration of the GPU system you have used to train the sentence bert and roberta models?

**Comment by nreimers** [comment]
They were performed on different systems. Most frequent I use most of the time a DGX-2 server.

**Comment by iamlockelightning** [comment]
Hi, @nreimers , is there any update on the multi-GPU training support?

**Comment by nreimers** [comment]
@iamlockelightning Sadly no progress.

Huggingface recently introduced:
https://github.com/huggingface/accelerate

Looks like a promising candidate to integrate here to enable multi GPU support

**Comment by HamedBabaei** [comment]
Hi, @nreimers, is there any progress on this?

**Comment by shoegazerstella** [comment]
Hi, is there any progress on this matter?

**Comment by xhluca** [comment]
The last release for this library was in June 2022. 

I am curious if there's any newly planned development on this, since multi-GPU training is a increasingly relevant; for instance, higher batch size matters more than traditional training (since gradient accumulation does not improve in-batch negative sampling, for example).

It seems that multi-gpu training has already been proposed here https://github.com/UKPLab/sentence-transformers/pull/1215, but there does not seem to have maintainers with the bandwidth to review this PR at UKP.

If there's no active development, is there plan in increasing the number of maintainers: by adding collaborators from other universities, [sponsoring community developers via github sponsors](https://github.com/sponsors), or receive support from foundations like LFAI? 

I believe sentence-transformers is an important open-source library for embeddings/retrieval. Without it, we'd be left with only OpenAI/Cohere/other commercial solutions; so I'd love to see more active development for the library.
